src/main/cs/RLBotDotNet/RLBotDotNet/Manager/BotManager.cs

```diff
@@ -19,16 +19,27 @@ namespace RLBotDotNet
     public class BotManager<T> where T : Bot
     {
         private readonly ConcurrentDictionary<int, BotLoopRenderer> _renderers;
-        private AutoResetEvent botRunEvent = new AutoResetEvent(false);
         private List<BotProcess> botProcesses = new List<BotProcess>();
         private Thread serverThread;
+
+        private readonly int frequency;
         
         /// <summary>
         /// Constructs a new instance of BotManager.
         /// </summary>
         public BotManager()
         {
             _renderers = new ConcurrentDictionary<int, BotLoopRenderer>();
```

Comment:
Is this a good opportunity for constructor chaining? https://stackoverflow.com/questions/1814953/c-sharp-constructor-chaining-how-to-do-it

Reply:
Thanks for the link! Didn't know about that. Changed it and also added a bounds check on the frequency to stop errors from occuring later down the line.